akka-remote/src/main/scala/akka/remote/artery/compress/CompressionTable.scala

```diff
@@ -26,7 +26,7 @@ private[akka] final case class CompressionTable[T](version: Int, map: Map[T, Int
 
       val vals = map.toList.sortBy(_._2).iterator.map(_._1)
       val dtab = Array.ofDim[Object](map.size).asInstanceOf[Array[T]]
-      vals.copyToArray(dtab) // TODO HEAVY, AVOID COPYING AND THE MAP ETC!!!
+      vals.copyToArray(dtab) // TODO rater heavy, can we optimise?
```

Comment:
Maybe side effect the copy each value when anyways iterating through it in the code on line 27?